### PR TITLE
Fix uid mismatch on /dev/kvm

### PR DESCRIPTION
**Pull request by agowa**
Fix virt-install failing if host and container use different uids for the kvm user.
e.g. virt-install throwing `ValueError: Host does not support domain type kvm for virtualization type 'hvm' arch 'x86_64'`

### CODE REVIEW

File: libvirtd.sh [hunk 1]
```diff
@@ -16,6 +16,9 @@ mkdir /dev.container && {
   keep pts
   mount --rbind /dev/pts/ptmx /dev/ptmx
   # Use the container /dev/kvm if available
+  # Replace inherited permissions from host with ours, as we have a different passwd file
+  # we may use different uids, e.g. kvm on the host is not kvm within the container.
+  [[ -e /dev.container/kvm ]] && chown root:kvm /dev.container/kvm && chmod 660 /dev.container/kvm
```
Comment (jean-edouard) [inline libvirtd.sh]: `/dev/kvm` is usually chmod-ed 666, which is probably a good idea here too, especially if we ever switch to non-root libvirt.

Comment (agowa) [reply libvirtd.sh]: it is? On my test machine it was set to 660 by default.
Do you have a source for that?
Also for non-root libvirt a group membership of that user within the "kvm" group is enough.

Comment (crobinso) [reply libvirtd.sh]: 50-udev-defaults from systemd sets this on modern distros, but the permissions look configurable at systemd build time: https://github.com/systemd/systemd/blob/master/rules.d/50-udev-default.rules.in#L86

On Fedora it's `0666` and RHEL too IIRC but I didn't confirm

Comment (agowa) [reply libvirtd.sh]: But why exactly does then a "kvm" group exist than? Isn't that group than completely redundant as everybody has access anyway?
Couldn't the file not just be owned by root:root with 0666 as permission??

Comment (andreabolognani) [reply libvirtd.sh]: I don't claim to understand the full implications of the various permission bits, but please keep in mind that SELinux is part of the picture too:

```
$ ls -lZ /dev/kvm 
crw-rw-rw-. 1 root kvm system_u:object_r:kvm_device_t:s0 10, 232 Jun 26 12:28 /dev/kvm
```

Comment (crobinso) [reply libvirtd.sh]: 'kvm' group predates widespread 0666 permissions on /dev/kvm. Also for a long time distro qemu packages were adding udev rules to set the 0666 permissions (before those rules were moved to systemd), so the host default was still 0660 unless qemu was installed. I believe the kvm group is just a historical artifact at least for distros that are setting 0666 permissions.

Comment (agowa) [reply libvirtd.sh]: Apparently the default by now is 0666: https://github.com/systemd/systemd/blob/master/meson_options.txt#L218-L219 and https://github.com/systemd/systemd/blob/master/meson.build#L797-L799

**Comment by rmohr** [comment]
Let me just add that in kubevirt/kubevirt we don't use this script, so changes there don't have any effect on kubevirt/kubevirt and can therefore be done savely without side-effects.

**Pull request closed without merging**